File: lib/sequencer/unit/import/zendesk/object_attribute/attribute_type/multiselect.rb
# Copyright (C) 2012-2025 Zammad Foundation, https://zammad-foundation.org/

class Sequencer::Unit::Import::Zendesk::ObjectAttribute::AttributeType::Multiselect < Sequencer::Unit::Import::Zendesk::ObjectAttribute::AttributeType::Select
  def data_type(...)
    'multiselect'
  end
end

File: app/frontend/shared/entities/object-attributes/form/resolver/fields/multiselect.ts
// Copyright (C) 2012-2026 Zammad Foundation, https://zammad-foundation.org/

import type { FieldResolverModule } from '#shared/entities/object-attributes/types/resolver.ts'

import { FieldResolverSelect } from './select.ts'

export class FieldResolverMultiselect extends FieldResolverSelect {
  // Multi-value selects ship as `contains one` in the advanced filter —
  // matches the existing overview / trigger condition vocabulary and the
  // backend selector wiring for multi-value attributes.
  protected override filterOperatorName = 'contains one'
}

export default <FieldResolverModule>{
  type: 'multiselect',
  resolver: FieldResolverMultiselect,
}
